Source/WebCore/editing/Editing.cpp

```diff
@@ -906,7 +906,7 @@ int caretMaxOffset(const Node& node)
     // For rendered text nodes, return the last position that a caret could occupy.
     if (is<Text>(node)) {
         if (auto* renderer = downcast<Text>(node).renderer())
-            return renderer->caretMaxOffset();
+            return std::min<int>(renderer->caretMaxOffset(), downcast<Text>(node).length());
```

Comment:
I understand that the renderer offsets do not all match text node offsets when we generate transformed text. But I don’t understand why _clamping_ is the correct way to convert a transformed text offset into an original text offset. Wouldn’t we sometimes be off when the error is not simply being too far past the end?

Instead it seems we would need to map in a way that understands the relationship between the transformed text and the original text based on how it was generated. Even if we only clamp for now as a partial solution, I would want to put the clamping in a place that knows the specifics of the text generation and could be fixed in the future. That probably means a renderer member function.

I also think we should seek out a test case that relies on these offsets being correct rather than just avoiding assertions.